File: ALODWebUtility/Reports/QueryCondition.vb
Namespace Reports

    Public Class QueryCondition

        Private _filter As String = ""
        Private _name As String = ""
        Private _type As DbType
        Private _value As Object

        Public Sub New(ByVal name As String, ByVal value As Object, ByVal valueType As DbType, ByVal filter As String)
            _name = name
            _type = valueType
            _value = value
            _filter = filter
        End Sub

        Public Sub New(ByVal name As String, ByVal filter As String)
            _name = name
            _filter = filter
        End Sub

        Public Property Field() As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                _name = value
            End Set
        End Property

        Public Property Filter() As String
            Get
                Return _filter
            End Get
            Set(ByVal value As String)
                _filter = value
            End Set
        End Property

        Public Property Name() As String
            Get
                Return "@" + _name
            End Get
            Set(ByVal value As String)
                _name = value
            End Set
        End Property

        Public Property Value() As Object
            Get
                Return _value
            End Get
            Set(ByVal value As Object)
                _value = value
            End Set
        End Property

        Public Property ValueType() As DbType
            Get
                Return _type
            End Get
            Set(ByVal value As DbType)
                _type = value
            End Set
        End Property

    End Class

End Namespace

File: ALODWebUtility/Reports/QueryCondition.cs
using System.Data;

namespace ALODWebUtility.Reports
{
    public class QueryCondition
    {
        private string _filter = "";
        private string _name = "";
        private DbType _type;
        private object _value;

        public QueryCondition(string name, object value, DbType valueType, string filter)
        {
            _name = name;
            _type = valueType;
            _value = value;
            _filter = filter;
        }

        public QueryCondition(string name, string filter)
        {
            _name = name;
            _filter = filter;
        }

        public string Field
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public string Filter
        {
            get
            {
                return _filter;
            }
            set
            {
                _filter = value;
            }
        }

        public string Name
        {
            get
            {
                return "@" + _name;
            }
            set
            {
                _name = value;
            }
        }

        public object Value
        {
            get
            {
                return _value;
            }
            set
            {
                _value = value;
            }
        }

        public DbType ValueType
        {
            get
            {
                return _type;
            }
            set
            {
                _type = value;
            }
        }
    }
}
